docs/source/assets_writer.rst

```diff
@@ -17,98 +17,112 @@
 
 .. code-block:: bash
 
-   # Finish the first trajectory. No further data will be accepted on it.
    rosservice call /finish_trajectory 0
-
-   # Ask Cartographer to serialize its current state.
    rosservice call /write_state ${HOME}/Downloads/b3-2016-04-05-14-14-00.bag.pbstream
 
-Now we run the assets writer with the `sample configuration file`_ for the 3D backpack:
 
-.. _sample configuration file: https://github.com/googlecartographer/cartographer_ros/blob/44459e18102305745c56f92549b87d8e91f434fe/cartographer_ros/configuration_files/assets_writer_backpack_3d.lua
+Once you've retrieved your ``.pbstream`` file, you can run the assets writer with the `sample pipeline`_ for the 3D backpack:
+
+.. _sample pipeline: https://github.com/googlecartographer/cartographer_ros/blob/44459e18102305745c56f92549b87d8e91f434fe/cartographer_ros/configuration_files/assets_writer_backpack_3d.lua
 
 .. code-block:: bash
 
    roslaunch cartographer_ros assets_writer_backpack_3d.launch \
       bag_filenames:=${HOME}/Downloads/b3-2016-04-05-14-14-00.bag \
       pose_graph_filename:=${HOME}/Downloads/b3-2016-04-05-14-14-00.bag.pbstream
 
-All output files are prefixed by ``--output_file_prefix`` which defaults to the filename of the first bag.
+All output files are prefixed with ``--output_file_prefix`` which defaults to the filename of the first bag.
 For the last example, if you specify ``points.ply`` in the pipeline configuration file, this will translate to ``${HOME}/Downloads/b3-2016-04-05-14-14-00.bag_points.ply``.
 
 Configuration
 -------------
 
-The assets writer is modeled as a pipeline.
-It consists of `PointsProcessor`_\ s and `PointsBatch`_\ s  flow through it.
-Data flows from the first processor to the next, each has the chance to modify the ``PointsBatch`` before passing it on.
+The assets writer is modeled as a pipeline of `PointsProcessor`_.
+`PointsBatch`_\ s flow through each processor and they all have the chance to modify the ``PointsBatch`` before passing it on.
 
 .. _PointsProcessor: https://github.com/googlecartographer/cartographer/blob/30f7de1a325d6604c780f2f74d9a345ec369d12d/cartographer/io/points_processor.h
 .. _PointsBatch: https://github.com/googlecartographer/cartographer/blob/30f7de1a325d6604c780f2f74d9a345ec369d12d/cartographer/io/points_batch.h
 
-For example the `assets_writer_backpack_3d.lua`_ uses ``min_max_range_filter`` to remove points that are either too close or too far from the sensor.
-After this, it writes X-Rays, then recolors the ``PointsBatch``\ s depending on the sensor frame ids and writes another set of X-Rays using these new colors.
+For example the `assets_writer_backpack_3d.lua`_ pipeline uses ``min_max_range_filter`` to remove points that are either too close or too far from the sensor.
+After this, it saves "*X-Rays*" (side views of the map), then recolors the ``PointsBatch``\ s depending on the sensor frame ids and writes another set of X-Rays using these new colors.
```

Comment:
an x-ray is not just a side view

Reply:
“Xray” in this context refers to being able to see through walls, i.e. such a render.

Reply:
“Xray” in this context refers to being able to see through walls, i.e. such a render.